src/kaleidoscope/plugin/LEDModeInterface.h

```diff
@@ -16,13 +16,32 @@
 
 #pragma once
 
+#include <Arduino.h>  // for PROGMEM
+
+#include "kaleidoscope/event_handler_result.h"  // for EventHandlerResult
+#include "kaleidoscope/event_handlers.h"        // for LedModeCallback
+
 namespace kaleidoscope {
 namespace plugin {
 
 class LEDModeInterface {
  public:
+  LEDModeInterface()
+    : led_mode_name_{0} {}
+  explicit LEDModeInterface(const char *led_mode_name)
+    : led_mode_name_{led_mode_name} {}
   void activate();
 
+  EventHandlerResult onLedEffectQuery(LedModeCallback callback) {
+    if (led_mode_name_ == 0) {
+      // If no name was defined, return a default string
+      callback("[unnamed led mode]");
```

Comment:
What's the case when we would want both this  string *and* the protected const char?

Reply:
I'm not sure I understand what you mean. Do you mean that I should assign this as the default value to the protected const char?

Reply:
Oh, right. Now that I'm looking at the diff I see what you mean. It used to be that I didn't assign a default value, which would cause this clause to be entered. I changed the default value to "[unnamed led mode]" at some point during my tests to see if I could reduce the size and it seems I forgot to remove the default value.
I'm happy with either option, though having a default empty value and this clause might make it move obvious what happens. Just let me know what you prefer.

Reply:
I'd probably default to an uninitialized value as the default and sending that string if the value is empty. It make it easier for other future code to reason about empty values.